src/OrchardCore/OrchardCore.Admin.Abstractions/AdminAttribute.cs

```diff
@@ -37,26 +37,54 @@ public class AdminAttribute : Attribute, IAsyncResourceFilter
         /// </summary>
         public string RouteName { get; set; }
 
-        public AdminAttribute(string template = null, string routeName = null)
+        public bool RequireAccessAdminPanelPermission { get; set; } = true;
```

Comment:
@Piedone There was no complain when we added the `Template` and `RouteName` parameters few weeks back. I don't think this is a good reason. `RequireAccessAdminPanelPermission` is set to true by default and this give us full control on the behavior of Admin attribute.

Reply:
The routing changes extended on top of the attribute's behavior, while this one alters the existing one. I understand that it's technically not breaking, but it does open up the option (what some features also utilize) to break a lot of assumptions.

Reply:
This is still an extension. It's an extension that allows you to control the underlining behavior. I think this is an okay change. It does not complicate the code by much. I like it

Reply:
`AdminAttribute` has implications well beyond the class where it is used, that's why this is an issue.

Until now, I as a module author could assume that `AdminAttribute.IsApplied()` being `true` means that the current request is for the admin area, accessed by admin users, who have a fundamentally different relationship with the website than other users. Now, this assumption can be broken, because it may be `true` for non-admin users as well, or even for anonymous users.

E.g. if I used this to display a notification or a widget that was meant to be seen on the admin, by admins, this is now not necessarily true.

In contrast, the routing changes didn't change any existing behavior or assumption of `AdminAttribute`. And if you choose to use it, it only changes behavior for the given class/action where it was used.

Reply:
I understand your point. But, I think in that case it's incorrect use case to use IsApplied() to assume the permission is met. IsApplied() indicates that the Attribute is applied or not. We could document this behavior change since we are releasing 2.0 and allow for this.

The problem with your recommended approach is that we are adding more filters (which is evaluated on every request) this cost can add up quickly.

Reply:
@Piedone Alternatively, we can add a parameter to `IsApplied()` `requireAccessAdminPanelPermission = true‘ this way we return true only when the attribute is applied and RequireAccessAdminPanelPermission is true.

Reply:
It's not just about the permissions, per se, but the completely different target audience. Currently: `AdminAttribute.IsApplied() == true` -> this is for people who work on this site. `AdminAttribute.IsApplied() == false` -> everybody else. After this: `AdminAttribute.IsApplied() == true` -> anybody. `AdminAttribute.IsApplied() == false` -> anybody. Keep in mind that checking `AdminAttribute.IsApplied()` is _the_ way to check if the current request is for the admin area. It's a crucial API.

And again, the change is not even necessary, and a problem better solved by the change password and login screens: they have their own layout by default (all share `Layout-Login`), and have an option to use the site theme. Doing this is not only in line what we do for other features already, and lacks any of the issues of the current implementation, but is also better in straightforward practical terms: that way, you have the option to keep those features design-wise (theming-wise) part of the frontend. This is necessary if you want to use e.g. 2FA on a public website with more than a handful of non-admin users (like we intend to do on DotNest), because then (just as login or registration) you want to style it with the site theme. Starting to set up 2FA from the frontend and ending up on a page styled with the admin theme and the empty menu, see below, is a disjointed user experience.

![image](https://github.com/OrchardCMS/OrchardCore/assets/1976647/9e09bf5b-b366-4054-b9f8-8e4bd21da3ac)

Reply:
@Piedone I made the requested change. Now we use `Layout-TwoFactor` for the 2FA views.

Reply:
Thank you!